### ISSUE TITLE
barakoa spawning block not enforced

### ISSUE THREAD

**Issue by AechtRob** [BobMowzie/MowziesMobs#270]
I have a  config of this:

barakoa {

    spawn_data {
        S:allowed_blocks <
            sand
         >

It used to work. With the latest update the parties of the b****** spawn anywhere within the required biome  :/

MC 1.12.2. Mod version **mowziesmobs-1.5.8**

**Comment by BobMowzie** [comment]
Do you have any other mods installed?

**Comment by AechtRob** [comment]
Yes, 80+ of them. It all worked fine until the new release though, although 2-3 mods also got updated at the same time. I can try and see if it is a conflict, but my hunch is not at the moment.

**Comment by AechtRob** [comment]
OK - I've just tested this with only MowziesMobs installed and the problem persists. I'm in 1.12 and this is the config I have: https://pastebin.com/ze3Jr1n6

And the things spawn in the biome but not restricted to the "sand" block. Has the way of referring to a block been changed? i.e. do we need to specify "minecraft:sand" now? If so, how to refer to meta and wildcards for meta?

Evidence (!): https://ibb.co/BjGWK91

**Comment by BobMowzie** [comment]
No, “sand” should work. Are you sure they aren’t spawning on sand and just walking away? They chase animals and can sometimes move very far from their spawn positions to get the kill.

**Comment by AechtRob** [comment]
In that picture, they spawned right there, on that savanna - I watched them pop up there.

**Comment by AechtRob** [comment]
Just updating this. I tried a biome blacklist on the barako raidingparty:

biomesoplenty:outback

But the continue to spawn in that biome.

**Comment by BobMowzie** [comment]
That sounds like a separate issue. I actually think I found the solution to the block problem. I’ll patch it soon.

As for this issue, are you sure that’s the name and path of the biome? Check your game log for any messages relating to it. It will mention something if the blacklist failed to find that biome.

**Comment by AechtRob** [comment]
Hi - I've double-checked the biome registry and it is definitely biomesoplenty:outback

I checked the debug log and it does have these lines:

**[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:185]: Mowzie's Mobs config warning: No biome with name 'biomesoplenty:outback'. Unable to blacklist.
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed
[14:17:14] [Client thread/INFO] [STDOUT]: [com.bobmowzie.mowziesmobs.server.spawn.SpawnHandler:getBiomesFromConfig:146]: Reflection failed**

Is the format of the biome name wrong? Does it need to go in as "Outback" (the actual name) or should we use the registry name (bimesoplenty:outback)?

Thanks!

**Comment by BobMowzie** [comment]
Oh wait what version of Mowzie’s Mobs are you using?

**Comment by AechtRob** [comment]
1.5.8 The biome issue looks like someone else reported something very similar just recently.

**Comment by BobMowzie** [comment]
That’s right. They were using an outdated version of the mod.

Does the issue go away if Mowzie’s Mobs and LLibrary are installed by themselves?
